### PR TITLE
issue-1137: Add redirect plugin

### LINKED ISSUE #1137: Add redirect plugin
## Feature Request

Add a redirect plugin and redirects to the Soul Gateway's own interface if the rules match

#### Additional context

Please read the [Contribution Guideline](https://dromara.org/website/en-us/docs/soul/contributor.html) before submitting the PR

### PR DESCRIPTION
**Pull request by HoldDie**
add redirect plugin.

Fixes #1137 

Make sure that:

- [x] You have read the [contribution guidelines](https://dromara.org/en-us/docs/soul/contributor.html).
- [x] You submit test cases (unit or integration tests) that back your changes.
- [ ] Your local test passed `mvn clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

File: soul-plugin/soul-plugin-redirect/src/main/java/org/dromara/soul/plugin/redirect/RedirectPlugin.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        return PluginEnum.REDIRECT.getCode();
+    }
+
+    @Override
+    public String named() {
+        return PluginEnum.REDIRECT.getName();
+    }
+
+    @Override
+    public Boolean skip(ServerWebExchange exchange) {
```
Comment (yu199195) [inline soul-plugin/soul-plugin-redirect/src/main/java/org/dromara/soul/plugin/redirect/RedirectPlugin.java:59]: remove this

File: soul-plugin/soul-plugin-redirect/src/main/java/org/dromara/soul/plugin/redirect/RedirectPlugin.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+import java.util.Objects;
+
+/**
+ * Redirect Plugin.
+ *
+ * @author HoldDie
+ */
+@Slf4j
+public class RedirectPlugin extends AbstractSoulPlugin {
+    @Override
```
Comment (yu199195) [inline soul-plugin/soul-plugin-redirect/src/main/java/org/dromara/soul/plugin/redirect/RedirectPlugin.java:48]: keep empty line

File: soul-plugin/soul-plugin-redirect/src/test/java/org/dromara/soul/plugin/redirect/RedirectPluginTest.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    }
+
+    @Test
+    public void testNamed() {
+        final String result = redirectPlugin.named();
+        assertThat(PluginEnum.REDIRECT.getName(), Matchers.is(result));
+    }
+
+    @Test
+    public void testSkip() {
```
Comment (yu199195) [inline soul-plugin/soul-plugin-redirect/src/test/java/org/dromara/soul/plugin/redirect/RedirectPluginTest.java:87]: remove this test

File: soul-spring-boot-starter/soul-spring-boot-starter-plugin/soul-spring-boot-starter-plugin-redirect/pom.xml [hunk 4]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (yu199195) [inline soul-spring-boot-starter/soul-spring-boot-starter-plugin/soul-spring-boot-starter-plugin-redirect/pom.xml:1]: add lincense

File: soul-spring-boot-starter/soul-spring-boot-starter-plugin/soul-spring-boot-starter-plugin-redirect/src/main/resources/META-INF/spring.factories [hunk 5]
```diff
@@ -0,0 +1,2 @@
+org.springframework.boot.autoconfigure.EnableAutoConfiguration=\
```
Comment (yu199195) [inline soul-spring-boot-starter/soul-spring-boot-starter-plugin/soul-spring-boot-starter-plugin-redirect/src/main/resources/META-INF/spring.factories:1]: add lincense

**Comment by yu199195** [comment]
please check ci

File: soul-plugin/soul-plugin-redirect/src/main/java/org/dromara/soul/plugin/redirect/RedirectPlugin.java (hunk 1 above)
Comment (HoldDie) [reply soul-plugin/soul-plugin-redirect/src/main/java/org/dromara/soul/plugin/redirect/RedirectPlugin.java]: down.

File: soul-plugin/soul-plugin-redirect/src/main/java/org/dromara/soul/plugin/redirect/RedirectPlugin.java (hunk 2 above)
Comment (HoldDie) [reply soul-plugin/soul-plugin-redirect/src/main/java/org/dromara/soul/plugin/redirect/RedirectPlugin.java:45]: down.

File: soul-plugin/soul-plugin-redirect/src/test/java/org/dromara/soul/plugin/redirect/RedirectPluginTest.java (hunk 3 above)
Comment (HoldDie) [reply soul-plugin/soul-plugin-redirect/src/test/java/org/dromara/soul/plugin/redirect/RedirectPluginTest.java]: down.

File: soul-spring-boot-starter/soul-spring-boot-starter-plugin/soul-spring-boot-starter-plugin-redirect/pom.xml (hunk 4 above)
Comment (HoldDie) [reply soul-spring-boot-starter/soul-spring-boot-starter-plugin/soul-spring-boot-starter-plugin-redirect/pom.xml:1]: down.

File: soul-spring-boot-starter/soul-spring-boot-starter-plugin/soul-spring-boot-starter-plugin-redirect/src/main/resources/META-INF/spring.factories (hunk 5 above)
Comment (HoldDie) [reply soul-spring-boot-starter/soul-spring-boot-starter-plugin/soul-spring-boot-starter-plugin-redirect/src/main/resources/META-INF/spring.factories:18]: down.

**Pull request merged**